show/main.py

```diff
@@ -393,6 +393,49 @@ def run_command_in_alias_mode(command):
         sys.exit(rc)
 
 
+def get_bgp_summary_extended(command):
+    """
+    Adds Neighbor name to the show ip[v6] bgp summary command
+    :param command: command to get bgp summary
+    """
+    try:
+        p = subprocess.Popen(command, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        device_output = p.stdout.read()
+        modified_output = []
+        my_list = iter(device_output.splitlines())
+        for element in my_list:
```

Comment:
You can simplify logic. and avoid nested for loops. 

import re
data_list = data.splitlines()

new_data = []

for i in data_list:
    if i.startswith("Neighbor "):
        new_i = "{}\tNeighborName".format(i)
        new_data.append(new_i)
    elif re.match(r"([0-9A-Fa-f]{1,4}:?|\d+\.\d+\.\d+.\d+)", i.split(" ")[0]):
        new_data.append(i + "\tNameOfNeighbor")
    elif i.startswith("Total number "):
        new_data.append(i)
    else:
        new_data.append(i)
        
for i in new_data:
    print(i)

Reply:
The command output lines that we modify are not always sequential. 
for eg. 
sometimes the ipv6 summary output have ipv6 in one line and rest parameters comes in another line.
```
Neighbor        V         AS MsgRcvd MsgSent   TblVer  InQ OutQ Up/Down  State/PfxRcd
afc0::20:1a2:1:1a
                4 64802     506     504        0    0    0 08:17:29        7
```
So used iterator and nested for loops. 
will check if I can simplify this.

Reply:
removed nested for loops in next iteration